src/lib/fixed_string.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+// FixedString is a data type, which stores a string in an array of chars in order to
+// save memory space by avoiding SSO
+class FixedString {
+ public:
+  // Create a FixedString from a std::string
+  // We did not find any good solution for creating a char array on stack
+  explicit FixedString(const std::string& string) : _mem(new char[string.size()]{}), _string_length(string.size()) {
+    std::memcpy(_mem, string.c_str(), _string_length);
+  }
+
+  // Create a FixedString from a memory address
+  FixedString(char* mem, size_t string_length) : _mem(mem), _string_length(string_length), _delete(false) {}
```

Comment:
The `_delete` flag acts as a semi-smart pointer implementation here, but you can create a case where the underlying string is deleted and then the pointer becomes invalid.

I see you use that constructor in `ValueVector::at()`. What is the data flow there, who "owns" the pointer / is that issue a real risk?

Reply:
Would be a nice idea, but does not work. When a FixedString is constructed from characters of a ValueVector and then gets deleted, the memory within the ValueVector will be freed. That is not the intended behaviour.